sentry-core/src/main/java/io/sentry/core/transport/AsyncConnection.java

```diff
@@ -83,6 +88,43 @@ public void send(final SentryEvent event, final @Nullable Object hint) throws IO
     executor.submit(new EventSender(event, hint, currentEventCache));
   }
 
+  // NOTE: This will not fallback to individual /store endpoints. Requires Sentry with Session
+  // health feature
+  @SuppressWarnings("FutureReturnValueIgnored")
+  @Override
+  public void send(@NotNull SentryEnvelope envelope, final @Nullable Object hint)
+      throws IOException {
+    // For now no caching on envelopes
+    //    IEventCache currentEventCache = NoOpEventCache.getInstance();
+    //    if (hint instanceof Cached) {
+    //      currentEventCache = NoOpEventCache.getInstance();
+    //    }
+
+    // Optimize for/No allocations if no items are under 429
+    List<SentryEnvelopeItem> dropItems = null;
+    for (SentryEnvelopeItem item : envelope.getItems()) {
+      if (item.getHeader() != null && transport.isRetryAfter(item.getHeader().getType())) {
+        if (dropItems == null) {
+          dropItems = new ArrayList<>();
+        }
+        dropItems.add(item);
+      }
+    }
+
+    if (dropItems != null) {
```

Comment:
Keep the same order otherwise session end could come before sessions start

Reply:
it doesn't matter as long as the sequence is right

Reply:
That's right